### DISCUSSION TITLE
Can 2 DevLake instances use the same database?

### DISCUSSION THREAD

**Discussion by GiesN** [apache/devlake#8149] [General]
Hello!
I am working on a setup of DevLake in an environment with 2 networks. The gitlab API is only reachable from a more secure network which means i need to install a 2nd instance of DevLake to get the data from there.
From this network i could reach the database of my main instance in the other network, which connects to AzureDevOps and GitHub.

Concluding this my question is, if 2 DevLake instances can use the same database or if there will be any problems when working together, like blocking access or in general working in parallel on the same db. Technically they for sure can connect to the same db, so the question is more on the level of how they access the database and if they will cause problems to each other.

Thanks in advance!

**Comment by klesh** [comment]
DevLake does NOT share the database because there is a blueprint scheduler, multiple instances would end up creating multiple pipelines for the same blueprint.
